connectors/filtering/validation.py

```diff
@@ -62,6 +64,16 @@ def valid_result(cls, rule_id):
             rule_id=rule_id, is_valid=True, validation_message="Valid rule"
         )
 
+    def __eq__(self, other):
+        if other is None:
```

Comment:
Mmm allowing `None`, but not other types is smelly -- also what happens if I compare with an `integer` for example?

I would be more drastic here and reject all tentative to compare for equality when `other` is not a `SyncRuleValidationResult` object:

```
if not isinstance(other, SyncRuleValidationResult):
    raise TypeError(f"Can't compare SyncRuleValidationResult with {type(other)}")
```

Reply:
Good point! Adjusted it.